jumanji/environments/packing/bin_pack/generator.py

```diff
@@ -1311,3 +1314,354 @@ def _unpack_items(self, state: State) -> State:
         )
         state.items_placed = jnp.zeros(placed_item_arr_dims, bool)
         return state
+
+
+class ExtendedTrainingGenerator(ExtendedRandomGenerator):
+    def __init__(
+        self,
+        max_num_items: int,
+        max_num_ems: int,
+        mean_item_value: float,
+        std_item_value: float,
+        min_target_volume: int,
+        max_target_volume: int,
+        split_eps: float = 0.3,
+        prob_split_one_item: float = 0.7,
+        split_num_same_items: int = 5,
+        container_dims: Tuple[int, int, int] = TWENTY_FOOT_DIMS,
+    ):
+        """
+        Args:
+            min_target_volume: minimal volume of the items in an instance in terms of containers.
+            min_target_volume: maximum volume of the items in an instance in terms of containers.
+        Raises:
+            ValueError: When the generator has to generate value based instances but the mean and
+            std aren't provided.
+        """
+        super().__init__(
+            max_num_items,
+            max_num_ems,
+            True,
+            True,
+            split_eps,
+            prob_split_one_item,
+            split_num_same_items,
+            container_dims,
+            mean_item_value,
+            std_item_value,
+        )
+        self.min_target_volume = min_target_volume
+        self.max_target_volume = max_target_volume
+
+    def _generate_value_based_solved_instance(  # type: ignore
+        self, key: chex.PRNGKey, target_volume: int
+    ) -> State:
+        """Generate the random instance with 1/target_volume of all items correctly packed. The
+        other part of the generated items remain unpacked.
+        """
+
+        key, split_key = jax.random.split(key)
+        container = make_container(self.container_dims)
+
+        list_of_ems = [container] + (self.max_num_ems - 1) * [empty_ems()]
+        ems = tree_transpose(list_of_ems)
+        ems_mask = jnp.zeros(self.max_num_ems, bool)
+
+        nb_items_in_one_container = math.floor(1 / target_volume * self.max_num_items)
+        items_spaces, final_items_mask = self._split_container_into_items_spaces(
+            container, split_key, nb_items_in_one_container
+        )
+        key, split_key = jax.random.split(key)
+        item_values = self.mean_item_value + (
+            self.std_item_value
+            * jax.random.normal(split_key, (len(final_items_mask),), jnp.float32)
+        )
+        len_inital_items_mask = final_items_mask.shape[0]
+        items = valued_item_from_space_and_max_value(items_spaces, item_values)
+        for _ in range(1, target_volume):
```

Comment:
I think this should be range(target_volume)

Reply:
it's not because items already contains the result of one call to `_split_container_into_items_spaces`